### ISSUE TITLE
edge version is triggering re-renders on nested pages

### ISSUE THREAD

**Issue by rvmourik** [nuxt/framework#6318]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.15.1`
- Nuxt Version:     `3.0.0-rc.6-27657653.000d81d`
- Package Manager:  `yarn@1.22.17`
- Builder:          `vite`
- User Config:      `head`, `ssr`, `buildModules`, `vite`, `runtimeConfig`, `build`
- Runtime Modules:  `-`
- Build Modules:    `@nuxtjs/tailwindcss@5.3.1`, `@vueuse/nuxt@9.0.2`, `@pinia/nuxt@0.3.1`
------------------------------

### Reproduction

I tried to reproduce it, but can't get it to fail in stackblitz.

### Describe the bug

When i updated to RC.6 I experienced no issues whatsoever, but updating to the latest edge all of the sudden my nested pages triggered two network requests for data. At first I thought the issue was with with the data fetching but after some debugging and placing a `console.log` in an `onMounted` hook I saw the `console.log` got triggered twice on client side navigating (initial load or refresh does not trigger it twice).

I have the following page structure:

```
layouts
  defaults.vue <-- has <slot>
pages
  workspace
    [workspace]
      dossier.vue  <-- fetched data with server api url (with $fetch)
  workspace.vue <-- has an async await in setup function and a `NuxtPage`
app.vue <-- has `NuxtLayout` and a `NuxtPage`
```
_http://localhost:3000/workspace/dynamicid/dossier_

I also tried disabling the date fetching in `dossier.vue` but the `onMounted` still gets called twice. When I disable the `await` in `workspace.vue` and let it run async the problem isn't there. Also placing an `onMounted` in workspace.vue get's called twice on navigating (and if i revert to rc.6 it does not).

All of the NuxtPage components do "not" have a page-key.

I think it has something to do with the combination of the top-level await in workspace.vue and maybe missing or present page-keys on the nested routes.

I am very sorry I can't provide you with a reproduction, but if you need any additional information I am happy to provide it to you.

Thanks in advance.

### Additional context

Maybe related items:

https://github.com/nuxt/framework/issues/3168
https://github.com/nuxt/framework/pull/4677
https://github.com/nuxt/framework/pull/6275

**Comment by danielroe** [comment]
@rvmourik Hopefully resolve in next edge release. Let me know if not.

**Comment by rvmourik** [comment]
@danielroe I already updated and it looks promising :-) Will test some more, if I find something I will let you know. Thanks for the quick follow up.

I don't now if it is related, but some other strange thing with routes is happening. I commented with some additional information in https://github.com/nuxt/framework/issues/6273.

Don't know if it is somehow related, but the issue occurs after updating to latest edge version.
